MIGRATION.md

```diff
@@ -864,6 +865,43 @@ import type { SvelteRenderer } from '@storybook/svelte';
 // etc.
 ```
 
+#### Renamed `DecoratorFn` to `Decorator`
+
+In 6.x you could import the type `DecoratorFn`:
+
+```ts
+import type { DecoratorFn } from '@storybook/react';
+// etc.
+```
+
+This type is deprecated in 7.0, instead you can use the type `Decorator`:
+
+```ts
+import type { Decorator } from '@storybook/react';
+// etc.
+```
+
+The type `Decorator` accepts a generic parameter `TArgs`. This can be used like this:
+
+```tsx
+import type { Decorator } from '@storybook/react';
+
+const withTheme: Decorator<{ theme: 'light' | 'dark' }> = (Story, { args }) => (
```

Comment:
I wonder if this example makes sense. We normally provide recipes using `parameters` for themes, given that in order for a story to have args for theming, you'd need to add global argTypes or something like that

Reply:
I see, what would be a better example? I personally used it, for providing mock services, or setting redux state.

Reply:
```tsx
import type { Decorator } from '@storybook/react';

const withLocale: Decorator<{ locale: 'en' | 'es' }> = (Story, { args }) => (
 <LocaleProvider lang={args.locale}>
    <Story />
  </LocaleProvider>
);
```